gtda/images/preprocessing.py

```diff
@@ -201,3 +201,115 @@ def transform(self, X, y=None):
         Xt = np.concatenate(Xt)
 
         return Xt
+
+
+@adapt_fit_transform_docs
+class ImageToPointCloud(BaseEstimator, TransformerMixin):
+    """Transformer returning a collection of point clouds that are the
+    coordinates in a 2- or 3-dimensional space of the activated pixel of the
+    input collection of 2-, respectively 3-dimensional binary images.
+    Deactivated pixels are given infinite coordinates in that space.
+
+    Notes
+    -----
+    This transformer is meant to transform a collection of imagea to a point
+    cloud so that collection of point clouds-based persistent homology module
+    can be applied.
+
+    Parameters
+    ----------
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1 unless
+        in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors.
+
+    Attributes
+    ----------
+    mesh_ : ndarray, shape (n_pixels_x * n_pixels_y [* n_pixels_z],
+        n_dimensions)
+        Mesh image for which each pixel value is its coordinates in a
+        `n_dimensions` space, where `n_dimensions` is the dimension of the
+        images of the input collection. Set in meth:`fit`.
+
+    See also
+    --------
+    gtda.homology.VietorisRipsPersistence, gtda.homology.SparseRipsPersistence,
+    gtda.homology.EuclideanCechPersistence
+
+    """
+
+    def __init__(self, n_jobs=None):
+        self.n_jobs = n_jobs
+
+    def _embed(self, X):
+        Xpts = np.stack([self.mesh_ for _ in range(X.shape[0])]) * 1.0
```

Comment:
I might have misunderstood the very idea of this transformer, but I am a little surprised that this is not just a simple-looking wrapper of `np.nonzero`. What am I missing?

Reply:
Let me look into it. It seems to me it could be.

Reply:
For now, this is the simplest solution. Once all persistent homology modules are equipped to handle list instead of arrays, then `np.nonzero` can be used. Right now, we set the coordinates of all deactivated points to be `np.inf`.

This will be handled in a future PR.

Reply:
@gtauzin I'm not sure I understand the reasoning for this, as I thought one could populate a pre-defined empty array using `np.nonzero`. In any case, we can merge this and open an issue for later.

Reply:
I created an issue.

Reply:
Yes, we could, but it is not worth the trouble as it is meant to evolve straight after.

Reply:
Cool!